sdk/python/tests/connectors/test_docker.py

```diff
@@ -45,3 +70,90 @@ async def connect_once():
         next(files)
     # assert the garbage was cleaned up
     assert not garbage_path.exists()
+
+
+def test_docker_cli_is_not_installed(
+    cache_dir: Path, mocked_image_ref, monkeypatch: pytest.MonkeyPatch
+):
+    """
+    When the docker cli is not installed ensure that the `FileNotFoundError` returned by
+    `subprocess.run` is wrapped by a `docker.ProvisionError` stating that
+    the command is not found.
+    """
+
+    def pached_subprocess_run(*args, **kwargs):
+        raise FileNotFoundError()
+
+    monkeypatch.setattr(docker.subprocess, "run", pached_subprocess_run)
```

Comment:
Using [pytest-subprocess](https://pytest-subprocess.readthedocs.io/en/latest/usage.html#basic-usage) can make this easier.